### DISCUSSION TITLE
Kırılan yumurtaları bulma sorusu

### DISCUSSION THREAD

**Discussion by sonsuzus** [sonsuzus/sonsuzus#7] [Zeka Sorusu]
Ahmet'in 2 özdeş yumurtası var ve bu yumurtaların dayanıklılığını test edecektir. Ahmet bu test ile 10 katlı bir binanın en düşük hangi katından atarsa bu yumurtaların kırılacağını bulmak istiyor. Her seferinde bir kattan bir tek yumurtayı yere atıyor ve kırılan bir yumurtayı bir daha atamıyor. En az kaç atışta yumurtaların en düşük kaçıncı kattan atılınca kırılacağını (ya da en yüksek kattan atılsa bile kırılamayacağını) kesin olarak bulabilir?

(Yumurtalar 5. kattan atılınca kırılıyorsa 5-10 arasındaki diğer tüm katlardan atılınca da kırılır.)

**Comment by HabilCcabbar** [comment]
Teknik olarak en düşük kaçıncı kattan kırılacağı tamamen şans. 11 ihtimal var: her kat için kırılması (10) veya hiçbir katta bir şekilde kırılmaması (11). Eğer en az kaç atışı arıyorsak en hızlı nasıl kırılır onu bulmamız lazım. Bunun içinde ilk kattan kırıldığını farz etmemiz lazım. (Eğer bu yumurtaları sadece katlardan düşürdüğünü farz edersek)

**Reply by sonsuzus** [reply]
Bunun için bir algoritma geliştirmelisin işte, soruda onu soruyor.

**Reply by HabilCcabbar** [reply]
Bu tarz mı hocam?
![Algoritma](https://github.com/sonsuzus/sonsuzus/assets/138613974/01fb7c25-293d-41b0-8c92-5b71a4298f97)

**Reply by sonsuzus** [reply]
Şema olarak doğru da bu şekilde yumurta 10. Katta kırılıyorsa bunu 10 denemede bulursun. Oysa daha kısa yöntemler var.

**Comment by AdaNilPoyraz** [comment]
İlk olarak en fazla katı nasıl eleyebileceğini düşünürsek 5. kattan atması gerekir. Eğer yumurta kırılırsa 6 - 7 - 8 - 9 -10. katlar elenir.
Eğer kırılmazsa 1 - 2 - 3 - 4 - 5. katlar elenir. Sonra sonuca bağlı olarak bir sonraki atacağı kat 3 veya 8 olur. böyle böyle hep kalan katların ortasındaki kattan atar.

**Reply by sonsuzus** [reply]
5 ten attın kırıldı, 3 den attın kırıldı. 

En yüksek limiti bulamadın ki?
